### PR TITLE
fix repo palette

### LINKED ISSUE #3568: Open repo command palette stopped working (F2)
Follows #3498

This is because fav bar and burger button with all repos was removed and so the source of repo list is not available anymore. Need to be refactored.

Will have a look.

### PR DESCRIPTION
**Pull request by sbcgua**
closes #3568

image

a lot of se80 reorderings so here are the changes
- js (remove old enumerator, add new one)
- chunk_lib->render_repo_palette
- removed render_scripts from page_main (was not used)
- add call of render_repo_palette to page_main and repo_view

build_main_menu, render_content, _add_col, has_favorites, set_filter - reordered, no modification (except register_deferred_script in render)

P.S. Appeared to be more tricky than expected with the current page structure ... certainly there is some space for refactoring

### CODE REVIEW

File: src/ui/zcl_abapgit_gui_page_main.clas.abap [hunk 1]
```diff
@@ -177,7 +163,7 @@ CLASS zcl_abapgit_gui_page_main IMPLEMENTATION.
 
         mv_repo_key = lv_key.
         CREATE OBJECT ei_page TYPE zcl_abapgit_gui_page_view_repo
-        EXPORTING iv_key = lv_key.
```
Comment (sbcgua) [inline src/ui/zcl_abapgit_gui_page_main.clas.abap:180]: @larshp this was not caught by abaplint 
https://github.com/abaplint/abaplint/issues/1028

**Comment by larshp** [comment]
@sbcgua check conflicts

**Comment by sbcgua** [comment]
should be OK now

File: src/ui/zcl_abapgit_gui_page_main.clas.abap (hunk 1 above)
Comment (larshp) [reply src/ui/zcl_abapgit_gui_page_main.clas.abap:180]: see comment in abaplint/abaplint#1028

**Pull request merged**